src/backtrace.ts

```diff
@@ -0,0 +1,183 @@
+/**
+ * @packageDocumentation
+ * Collects backtraces for executables for diagnostics
+ */
+import { stringifyError } from "./errors.js";
+import * as actionsCore from "@actions/core";
+import * as exec from "@actions/exec";
+import { readFile, readdir } from "node:fs/promises";
+import { promisify } from "node:util";
+import { gzip } from "node:zlib";
+import os from "os";
+
+export async function collectBacktraces(
+  prefixes: string[],
+): Promise<Map<string, string>> {
+  if (os.platform() === "darwin") {
+    return await collectBacktracesMacOS(prefixes);
+  }
+  if (os.platform() === "linux") {
+    return await collectBacktracesSystemd(prefixes);
+  }
+
+  return new Map();
+}
+
+export async function collectBacktracesMacOS(
+  prefixes: string[],
+): Promise<Map<string, string>> {
+  const backtraces: Map<string, string> = new Map();
+
+  try {
+    const { stdout: logJson } = await exec.getExecOutput(
+      "log",
+      [
+        "show",
+        "--style",
+        "json",
+        "--last",
+        // Note we collect the last 1m only, because it should only take a few seconds to write the crash log.
+        // Therefor, any crashes before this 1m should be long done by now.
+        "1m",
+        "--no-info",
+        "--predicate",
+        "sender = 'ReportCrash'",
+      ],
+      {
+        silent: true,
+      },
+    );
+
+    const sussyArray: unknown = JSON.parse(logJson);
+    if (!Array.isArray(sussyArray)) {
+      throw new Error(`Log json isn't an array: ${logJson}`);
+    }
+
+    if (sussyArray.length > 0) {
+      actionsCore.info(`Collecting crash data...`);
+      const delay = async (ms: number): Promise<void> =>
+        new Promise((resolve) => setTimeout(resolve, ms));
+      await delay(5000);
+    }
+  } catch (e: unknown) {
+    actionsCore.debug(
+      "Failed to check logs for in-progress crash dumps, assuming there are none.",
```

Comment:
Use a semicolon instead of a comma. Nitpicky, yes, but that makes the meaning of the "assuming there are none" clause more clear.

Reply:
Actually, let's go beyond that and make that a full sentence, maybe:

```
Failed to check logs for in-progress crash dumps; now proceeding with the assumption that there are no crash dumps
```